R/pkg/R/SQLContext.R

```diff
@@ -341,11 +342,13 @@ setMethod("toDF", signature(x = "RDD"),
 #' @name read.json
 #' @method read.json default
 #' @note read.json since 1.6.0
-read.json.default <- function(path) {
+read.json.default <- function(path, ...) {
   sparkSession <- getSparkSession()
+  options <- varargsToStrEnv(...)
   # Allow the user to have a more flexible definiton of the text file path
   paths <- as.list(suppressWarnings(normalizePath(path)))
   read <- callJMethod(sparkSession, "read")
+  read <- callJMethod(read, "options", options)
```

Comment:
Oh, I haven't tested this yet but the first `path` will have higher precedence because in `setWriteOptions`,

```
options <- varargsToStrEnv(...)
if (!is.null(path)) {
  options[["path"]] <- path
}
```

We are setting the first `path` later overwriting the existing path.

It seems this way is the same in Scala as well.

```
  def load(path: String): DataFrame = {
    option("path", path).load(Seq.empty: _*) // force invocation of `load(...varargs...)`
  }
```

It overwrites the path in the `option` when it actually loads. Let me test this and leave another comment in R.